kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala

```diff
@@ -59,14 +62,26 @@ class FlinkProcessBuilder(
   // flink.execution.target are required in Kyuubi conf currently
   val executionTarget: Option[String] = conf.getOption("flink.execution.target")
 
+  private lazy val proxyUserEnable: Boolean = {
+    doAsEnabled && conf.get(ENGINE_FLINK_DOAS_ENABLED)
```

Comment:
> ... it can only be enabled on Kerberized environment.

I think the logic here should be 
```
if (conf.get(ENGINE_FLINK_DOAS_ENABLED)) {
  if (!UserGroupInformation.isSecurityEnabled) {
     log warning message
     false
  } else {
     true
  }
}
```